src/mbgl/renderer/layers/render_heatmap_layer.cpp

```diff
@@ -368,9 +384,27 @@ void RenderHeatmapLayer::update(gfx::ShaderRegistry& shaders,
             /* .weight_t = */ std::get<0>(paintPropertyBinders.get<HeatmapWeight>()->interpolationFactor(zoom)),
             /* .radius_t = */ std::get<0>(paintPropertyBinders.get<HeatmapRadius>()->interpolationFactor(zoom)),
             /* .padding = */ {0}};
+        if (interpolationUniformBuffer) {
+            interpolationUniformBuffer->update(&interpolateUBO, sizeof(interpolateUBO));
+        } else {
+            interpolationUniformBuffer = context.createUniformBuffer(&interpolateUBO, sizeof(interpolateUBO));
+        }
+
+        const HeatmapEvaluatedPropsUBO evaluatedPropsUBO = {
+            /* .weight = */ uniformValue<style::HeatmapWeight>(paintPropertyBinders, evaluated),
+            /* .radius = */ uniformValue<style::HeatmapRadius>(paintPropertyBinders, evaluated),
+            /* .intensity = */ evaluated.get<HeatmapIntensity>(),
+            /* .padding = */ 0};
+        if (evaluatedPropsUniformBuffer) {
+            evaluatedPropsUniformBuffer->update(&evaluatedPropsUBO, sizeof(evaluatedPropsUBO));
+        } else {
+            evaluatedPropsUniformBuffer = context.createUniformBuffer(&evaluatedPropsUBO, sizeof(evaluatedPropsUBO));
+        }
```

Comment:
From what I know the evaluated props are uniform values and we don't need property binders to extract them. In case the values is a vertex attribute the uniform value it will be unused so it's not important what contains.